src/core/core.tooltip.js

```diff
@@ -221,52 +229,81 @@ function createTooltipItem(element) {
  * Helper to get the reset model for the tooltip
  * @param tooltipOpts {Object} the tooltip options
  */
-function getBaseModel(tooltipOpts) {
+function getBaseModel(tooltipOpts, chart) {
 	var globalDefaults = defaults.global;
-
-	return {
+	var context = {
+		chart: chart,
+	};
+	var defaultFontColor = 'defaultFontColor';
+	var defaultFontFamily = 'defaultFontFamily';
+	var defaultFontSize = 'defaultFontSize';
+	var defaultFontStyle = 'defaultFontStyle';
+	var DEFAULT_MAP = {
+		bodyFontColor: defaultFontColor,
+		bodyFontFamily: defaultFontFamily,
+		bodyFontSize: defaultFontSize,
+		bodyFontStyle: defaultFontStyle,
+		footerFontColor: defaultFontColor,
+		footerFontFamily: defaultFontFamily,
+		footerFontSize: defaultFontSize,
+		footerFontStyle: defaultFontStyle,
+		titleFontColor: defaultFontColor,
+		titleFontFamily: defaultFontFamily,
+		titleFontSize: defaultFontSize,
+		titleFontStyle: defaultFontStyle,
+	};
+	var i, ilen, key;
+	var keys = [
+		'bodyFontColor', 'bodyFontSize',
+		'titleFontColor', 'titleFontSize',
+		'titleSpacing', 'titleMarginBottom',
+		'footerFontColor', 'footerFontSize',
+		'footerSpacing', 'footerMarginTop',
+		'caretSize', 'cornerRadius', 'backgroundColor',
+		'displayColors', 'borderColor', 'borderWidth',
+	];
+	var underscoreKeys = [
+		'bodyFontFamily', 'bodyFontStyle',
+		'titleFontFamily', 'titleFontStyle',
```

Comment:
It's a bit inconsistent that some lines have 1 property and some lines have 2. I might standardize it on being 1 everywhere including for `keys` above

Reply:
Addressed this and sorted the arrays in order